test/metabase/test/data/crate.clj

```diff
@@ -0,0 +1,75 @@
+(ns metabase.test.data.crate
+  "Code for creating / destroying a Crate database from a `DatabaseDefinition`."
+  (:require [environ.core :refer [env]]
+            metabase.driver.crate
+            (metabase.test.data [generic-sql :as generic]
+                                [interface :as i])
+            [metabase.driver.generic-sql :as sql]
+            [clojure.java.jdbc :as jdbc]
+            [metabase.util :as u]
+            [clojure.string :as s])
+  (:import metabase.driver.crate.CrateDriver))
+
+(def ^:private ^:const field-base-type->sql-type
+  {:BigIntegerField "long"
+   :BooleanField    "boolean"
+   :CharField       "string"
+   :DateField       "timestamp"
+   :DateTimeField   "timestamp"
+   :DecimalField    "integer"
+   :FloatField      "float"
+   :IntegerField    "integer"
+   :TextField       "string"
+   :TimeField       "timestamp"})
+
+
+(defn- timestamp->CrateDateTime
+  [value]
+  (if (= (instance? java.sql.Timestamp value) true)
+    (.getTime (u/->Timestamp value))
+    (if (= (and (= (instance? clojure.lang.PersistentArrayMap value) true) (contains? value :korma.sql.utils/generated)) true)
+      (+ (read-string (s/replace (:korma.sql.utils/generated value) #"CURRENT_TIMESTAMP \+" "")) (.getTime (u/new-sql-timestamp)))
```

Comment:
@camsaul hm. I know this line looks really odd. However, I wasn't able to use the `korma.core/raw` function to generate the insert-value. I think it is related to plain `jdbc/insert!` method that I use to generate a non-transactional insert (Crate does not support transactions). I'm using it now like this:

```clojure
(defn- timestamp->CrateDateTime
  [value]
  (if (instance? java.sql.Timestamp value)
    (.getTime (u/->Timestamp value))
    (if (and (instance? clojure.lang.PersistentArrayMap value) (contains? value :korma.sql.utils/generated))
      (k/raw value)
      value)))
```

And get this stacktrace:
```
act-msg: exception in actual: (metabase.test.data.datasets/do-with-engine :crate actual--873848585)
    threw: class java.lang.ClassCastException - clojure.lang.Keyword cannot be cast to java.lang.String
           io.crate.shade.org.elasticsearch.common.io.stream.StreamOutput$writeGenericValue (StreamOutput.java:381)
           io.crate.action.sql.SQLRequest$writeTo (SQLRequest.java:96)
           io.crate.shade.org.elasticsearch.transport.netty.NettyTransport$sendRequest (NettyTransport.java:701)
           io.crate.shade.org.elasticsearch.transport.TransportService$sendRequest (TransportService.java:276)
           io.crate.shade.org.elasticsearch.action.TransportActionNodeProxy$execute (TransportActionNodeProxy.java:57)
           io.crate.client.InternalCrateClient$1$doWithNode (InternalCrateClient.java:92)
           io.crate.shade.org.elasticsearch.client.transport.TransportClientNodesService$execute (TransportClientNodesService.java:206)
           io.crate.client.InternalCrateClient$execute (InternalCrateClient.java:87)
           io.crate.client.InternalCrateClient$execute (InternalCrateClient.java:70)
           io.crate.client.InternalCrateClient$sql (InternalCrateClient.java:59)
           io.crate.client.CrateClient$sql (CrateClient.java:151)
           io.crate.client.jdbc.CratePreparedStatement$executeSingle (CratePreparedStatement.java:173)
           io.crate.client.jdbc.CratePreparedStatement$execute (CratePreparedStatement.java:163)
           io.crate.client.jdbc.CratePreparedStatement$executeUpdate (CratePreparedStatement.java:124)
           on (jdbc.clj:727)
           clojure.java.jdbc$db_do_prepared_return_keys$invokeStatic (jdbc.clj:742)
           on (jdbc.clj:714)
           on (jdbc.clj:937)
           clojure.java.jdbc$multi_insert_helper$invokeStatic (jdbc.clj:936)
           on (jdbc.clj:931)
           clojure.java.jdbc$insert_helper$invokeStatic (jdbc.clj:948)
           on (jdbc.clj:940)
           clojure.java.jdbc$insert_BANG_$invokeStatic (jdbc.clj:1039)
           clojure.java.jdbc$insert_BANG_$doInvoke (jdbc.clj:1024)
           on (crate.clj:48)
           on (generic_sql.clj:183)
           on (crate.clj:55)
           on (generic_sql.clj:22)
           metabase.test.data.generic_sql$create_db_BANG_$invokeStatic (generic_sql.clj:324)
           on (generic_sql.clj:300)
           on (interface.clj:64)
           metabase.test.data$get_or_create_database_BANG_$invokeStatic (data.clj:178)
           on (data.clj:167)
           metabase.test.data$do_with_temp_db$invokeStatic (data.clj:248)
           on (data.clj:243)
           metabase.query_processor_test$count_of_grouping$invokeStatic (query_processor_test.clj:1494)
           metabase.query_processor_test$count_of_grouping$doInvoke (query_processor_test.clj:1493)
           on (query_processor_test.clj:1509)
           metabase.test.data.datasets$do_with_engine$invokeStatic (datasets.clj:72)
           on (datasets.clj:69)
           user$eval83552$invokeStatic (form-init8592673036465478053.clj:1)
           on (form-init8592673036465478053.clj:1)
```

I think you can't use the `k/raw` function on plain clojure jdbc statements. Maybe you know how to do that properly?